src/main/java/com/zufar/onlinestore/payment/converter/PaymentIntentConverter.java

```diff
@@ -5,29 +5,39 @@
 import com.zufar.onlinestore.payment.calculator.PaymentPriceCalculator;
 import com.zufar.onlinestore.payment.dto.CreatePaymentDto;
 import com.zufar.onlinestore.payment.entity.Payment;
-import lombok.RequiredArgsConstructor;
-import org.springframework.stereotype.Component;
+import org.mapstruct.Mapper;
 
-@RequiredArgsConstructor
-@Component
-public class PaymentIntentConverter {
+@Mapper
+public interface PaymentIntentConverter {
 
-    private final PaymentPriceCalculator paymentPriceCalculator;
+    PaymentPriceCalculator paymentPriceCalculator = new PaymentPriceCalculator();
 
-    public Payment toPayment(final PaymentIntent paymentIntent) {
-        return Payment.builder()
-                .itemsTotalPrice(paymentPriceCalculator.calculatePriceForPayment(paymentIntent.getAmount()))
-                .paymentIntentId(paymentIntent.getId())
-                .currency(paymentIntent.getCurrency())
-                .build();
+    default Payment toPayment(final PaymentIntent paymentIntent) {
+        if (paymentIntent == null) {
+            return null;
```

Comment:
returning `null` is dangerous
throw new custom exception here with the detailed error description here
like `InvalidPaymentIntentException("PaymentIntent is null")`